test/common/user_dedicated_admin_synced_sso.go

```diff
@@ -431,15 +431,18 @@ func cleanUpTestDedicatedAdminUsersSyncedSSO(ctx context.Context, t TestingTB, c
 		t.Fatalf("failed to delete OpenShift user %s, err: %v", testUser.Name, err)
 	}
 
-	// Ensure KeycloakUser CR is deleted
-	err = c.Delete(ctx, &keycloak.KeycloakUser{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      fmt.Sprintf("%s-%s", TestingIDPRealm, testUserName),
-			Namespace: RHSSOProductNamespace,
-		},
+	// Ensure KeycloakUser CR has been deleted within 2 minutes
+	kcUser := &keycloak.KeycloakUser{}
+	err = wait.Poll(time.Second*15, time.Minute*2, func() (done bool, err error) {
+		err = c.Get(ctx, types.NamespacedName{Name: fmt.Sprintf("%s-%s", TestingIDPRealm, testUserName),
+			Namespace: RHSSOProductNamespace}, kcUser)
+		if err == nil {
+			return false, nil
+		}
```

Comment:
I think we should continue only if kc user cr is gone, so only continue if the err != nil && error is IsNotFound

Reply:
Reason for it is that the error could be for example for another reason, like interrupted connection for example